docs/community/release_guide.md

```diff
@@ -0,0 +1,291 @@
+```
+
+#### Java Home
+
+An available environment variable `JAVA_HOME`, you can do `echo $JAVA_HOME` to check it.
+Note that, the Java version should be 8.
+
+#### Subversion
+
+Besides on `git`, `svn` is also required for Apache release, please refer to
+https://www.apache.org/dev/version-control.html#https-svn for details.
+
+#### GPG Key
+
+You need to have a GPG key to sign the release artifacts. Please be aware of the ASF-wide
+[release signing guidelines](https://www.apache.org/dev/release-signing.html). If you don’t have a GPG key associated
+with your Apache account, please create one according to the guidelines.
+
+Determine your Apache GPG Key and Key ID, as follows:
+
+```shell
+gpg --list-keys --keyid-format SHORT
+```
+
+This will list your GPG keys. One of these should reflect your Apache account, for example:
+
+```shell
+pub   rsa4096/37356FA0 2022-10-27 [SC] [expires: 2028-10-27]
+      FCF20BB29C7BEFDF58F998F76392F71F37356FA0
+uid         [ultimate] Ethan Feng <ethanfeng@apache.org>
+sub   rsa4096/D2ED4CBF 2022-10-27 [E] [expires: 2028-10-27]
+```
+
+> Note: To follow the [Apache's release specification](https://infra.apache.org/release-signing.html#note), all new RSA keys generated should be at least 4096 bits. Do not generate new DSA keys.
+
+Here, the key ID is the 8-digit hex string in the pub line: `37356FA0`.
+
+To export the PGP public key, using:
+
+```shell
+gpg --armor --export 37356FA0
+```
+
+If you have more than one gpg key, you can specify the default key as the following:
+
+```
+echo 'default-key <key-fpr>' > ~/.gnupg/gpg.conf
+```
+
+The last step is to update the KEYS file with your code signing key
+https://www.apache.org/dev/openpgp.html#export-public-key
+
+```shell
+svn checkout --depth=files "https://dist.apache.org/repos/dist/release/incubator/celeborn/" work/svn-celeborn
+
+(gpg --list-sigs "${ASF_USERNAME}@apache.org" && gpg --export --armor "${ASF_USERNAME}@apache.org") >> work/svn-celeborn/KEYS
+
+svn commit --username "${ASF_USERNAME}" --password "${ASF_PASSWORD}" --message "Update KEYS" work/svn-celeborn
+```
+
+## Cut branch if for feature release
+
+celeborn use version pattern `{MAJOR_VERSION}.{MINOR_VERSION}.{PATCH_VERSION}`, e.g. `0.3.1`.
+__Feature Release__ means `MAJOR_VERSION` or `MINOR_VERSION` changed, and __Patch Release__ means `PATCH_VERSION` changed.
+
+The main step towards preparing a feature release is to create a release branch. This is done via standard Git branching
+mechanism and should be announced to the community once the branch is created.
+
+> Note: If you are releasing a patch version, you can ignore this step.
+
+The release branch pattern is `branch-{MAJOR_VERSION}.{MINOR_VERSION}`, e.g. `branch-1.7`.
+
+After cutting release branch, don't forget bump version in `master` branch.
+
+## Build a release candidate
+
+> Don't forget to switch to the release branch!
+
+- Set environment variables.
+
+```shell
+export RELEASE_VERSION=<release version, e.g. 0.3.1>
```

Comment:
Does this need to export `RELEASE_VERSION` for `RELEASE_VERSION=$(awk -F'"' '/ThisBuild \/ version/ {print $2}' version.sbt)` in `release.sh`?

Reply:
Yes. I'll change the doc.